tests/unit/test_languageHandler.py

```diff
@@ -31,3 +98,119 @@ def test_nonStandardLocale(self):
 		lcid = languageHandler.localeNameToWindowsLCID("zzzz")
 		self.assertEqual(lcid, LCID_NONE)
+
+
+class TestSetLocale(unittest.TestCase):
+	"""
+	tests setting the python locale for all possible locales set by NVDA or the System
+	"""
+
+	SUPPORTED_LOCALES = [("en", "en_US"), ("fa-IR", "fa_IR"), ("an-ES", "an_ES")]
+
+	@classmethod
+	def tearDownClass(cls):
+		languageHandler.setLanguage(languageHandler.curLang)
+
+	def setUp(self):
+		languageHandler.setLanguage(languageHandler.curLang)
+
+	def testSupportedLocales(self):
+		for localeName in self.SUPPORTED_LOCALES:
+			languageHandler.setLocale(localeName[0])
+			self.assertEqual(locale.getlocale()[0], localeName[1])
+
+	def testUnsupportedLocales(self):
+		original_locale = locale.getlocale()
+		for localeName in UNSUPPORTED_PYTHON_LOCALES:
+			with self.subTest():
+				languageHandler.setLocale(localeName)
+				self.assertEqual(locale.getlocale(), original_locale)
+
+	def testAllSupportedLangs(self):
+		for localeName in TRANSLATABLE_LANGS - UNSUPPORTED_PYTHON_LOCALES:
+			with self.subTest():
+				languageHandler.setLocale(localeName)
+				current_locale = locale.getlocale()
+
+				if localeName == "uk":
+					self.assertEqual(current_locale[0], "English_United Kingdom")
+				else:
+					pythonLang = current_locale[0].split("_")[0]
+					langOnly = localeName.split("_")[0]
+					self.assertEqual(
+						langOnly,
+						pythonLang,
+						f"full values: {localeName} {current_locale[0]}",
+					)
+
+	def testAllWindowsLangs(self):
+		prev_locale = locale.getlocale()
+		for localeName in WINDOWS_LANGS:
+			with self.subTest():
+				languageHandler.setLocale(localeName)
+				current_locale = locale.getlocale()
+				if localeName == languageHandler.curLang:
+					self.assertEqual(current_locale, prev_locale)
+				elif localeName in UNSUPPORTED_PYTHON_LOCALES:
+					self.assertEqual(current_locale, prev_locale)
+				else:
+					self.assertNotEqual(current_locale, prev_locale, localeName)
+
+
+class TestSetLanguage(unittest.TestCase):
+	"""
+	tests setting the NVDA language for all possible locales set by NVDA
+	"""
+	UNSUPPORTED_WIN_LANGUAGES = ["an", "kmr"]
+
+	def tearDown(self):
+		languageHandler.setLanguage(self._prevLang)
+
+	def __init__(self, *args, **kwargs):
+		self._prevLang = languageHandler.getLanguage()
+
+		ctypes.windll.kernel32.SetThreadLocale(0)
+		defaultThreadLocale = ctypes.windll.kernel32.GetThreadLocale()
+		self._defaultThreadLocaleName = languageHandler.windowsLCIDToLocaleName(
+			defaultThreadLocale
+		)
+
+		locale.setlocale(locale.LC_ALL, "")
+		self._defaultPythonLocale = locale.getlocale()
+
+		languageHandler.setLanguage(self._prevLang)
+		super().__init__(*args, **kwargs)
+
+	def testTranslatableLanguages(self):
+		for lang in TRANSLATABLE_LANGS:
+			langOnly = lang.split("_")[0]
+			with self.subTest():
+				languageHandler.setLanguage(lang)
+				# check curLang/translation service is set
+				self.assertEqual(languageHandler.curLang, lang)

```

Comment:
There seems to be three separate assert blocks, could you explain why?

I'd add some docs to the test something like:
>Tests languageHandler.setLanguage, which should do the following:
> - set the languageHandler current language
> - set the locale for the windows thread
> - set the python locale